Index: linux-5.4.92/drivers/gpio/Makefile
===================================================================
--- linux-5.4.92.orig/drivers/gpio/Makefile
+++ linux-5.4.92/drivers/gpio/Makefile
@@ -171,3 +171,4 @@ obj-$(CONFIG_GPIO_XTENSA)		+= gpio-xtens
 obj-$(CONFIG_GPIO_ZEVIO)		+= gpio-zevio.o
 obj-$(CONFIG_GPIO_ZX)			+= gpio-zx.o
 obj-$(CONFIG_GPIO_ZYNQ)			+= gpio-zynq.o
+obj-y += edgecore_reboot.o
